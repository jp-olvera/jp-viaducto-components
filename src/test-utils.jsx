import React from 'react';
import { render } from '@testing-library/react';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }) => {
  return <ConfigProvider>{children}</ConfigProvider>;
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
