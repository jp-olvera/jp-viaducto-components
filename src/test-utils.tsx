import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ConfigProvider } from './providers';

const AllTheProviders = ({ children }: any) => (
  <ConfigProvider>{children}</ConfigProvider>
);

jest.mock('./cells/Input/amex.svg', () => null);
jest.mock('./cells/Input/mastercard.svg', () => null);
jest.mock('./cells/Input/visa.svg', () => null);
jest.mock('./cells/Dropdown/sorting.svg', () => null);
jest.mock('./cells/BareButton/close.svg', () => null);

const customRender = (ui, options?) => render(ui, { wrapper: AllTheProviders, ...options });
// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
