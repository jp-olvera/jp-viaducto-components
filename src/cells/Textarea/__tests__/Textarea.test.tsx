/* eslint-env jest */

import React from 'react';
import { render, fireEvent } from '../../../test-utils';
import { Textarea } from '..';

describe('<TextArea /> component', () => {
  test('should render properly', () => {
    const { container } = render(<Textarea />);
    expect(container.querySelector('textarea')).not.toBeNull();
  });
  test('should render properly with props', () => {
    const { container } = render(
      <Textarea family={undefined} resizable radius='md' borderColor='red' />,
    );
    expect(container.querySelector('textarea')).not.toBeNull();
  });
  test('should render disabled textarea', () => {
    const { container } = render(<Textarea disabled />);
    expect(container.querySelector('textarea')).toMatchSnapshot();
  });
  test('should write in the textarea', () => {
    const { container } = render(
      <Textarea family={undefined} resizable radius='lg' />,
    );
    const textarea = container.querySelector('textarea');
    fireEvent.change(textarea || window, {
      target: { value: 'Im the new text' },
    });
    expect(textarea?.value).not.toBeNull();
  });
});
