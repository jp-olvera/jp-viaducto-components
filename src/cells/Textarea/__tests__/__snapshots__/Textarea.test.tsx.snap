// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> component should render disabled textarea 1`] = `
<textarea
  class="StyledTextarea-sc-1jfvl7f-0 ddRsYE"
  disabled=""
  rows="1"
/>
`;
